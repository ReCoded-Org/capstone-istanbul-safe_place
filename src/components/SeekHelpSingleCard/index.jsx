import React from "react";
import phoneSvg from "./images/phone.svg";
import locationSvg from "./images/map-marker.svg";
import doctorImg from "./images/doctorImage.svg";
import defaultImg from "./images/defaultImg.svg";
import "./index.scss";

const SeekHelpSingleCard = ({
  title,
  name,
  portraitImg,
  phoneNumber,
  address,
  moreInfo,
}) => {
  const imageMap = {
    doctorImg: doctorImg,
  };

  return (
    <>
      <div className="carouselItem" key={name}>
        <div className="carouselHeaderAndImage">
          <div>
            <h3>{title}</h3>
            <h4>{name}</h4>
          </div>
          <div>
            <img src={imageMap[portraitImg] || defaultImg} alt="Specialist" />
          </div>
        </div>
        <div>
          <div className="contactDetails">
            <div>
              <img src={phoneSvg} alt="Phone Icon" />
            </div>
            <div>
              <p>{phoneNumber}</p>
            </div>
          </div>
          <div className="contactDetails">
            <div>
              <img src={locationSvg} alt="Location Icon" />
            </div>
            <div>
              <p>{address}</p>
            </div>
          </div>
          <p>{moreInfo}</p>
        </div>
      </div>
    </>
  );
};

export default SeekHelpSingleCard;
