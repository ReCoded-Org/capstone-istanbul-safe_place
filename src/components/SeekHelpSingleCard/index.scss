@import "../../style/variables";

.singleCard {
  background-color: $warningColor;
  border-radius: 20px;
  padding: 2rem;

  h3 {
    color: $darkColor;
    font-size: 42px;
    font-weight: bold;
    font-family: $mainFontFamily;
  }
  h4 {
    color: $darkColor;
    font-size: 20px;
    font-family: $mainFontFamily;
  }
  p {
    color: $darkColor;
    font-size: 12px;
  }
}

.carouselHeaderAndImage {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.contactDetails {
  display: flex;
  flex-direction: row;
}
