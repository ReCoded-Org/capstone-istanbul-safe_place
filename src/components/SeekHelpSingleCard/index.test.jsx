import React from "react";
import renderer from "react-test-renderer";
import SeekHelpSingleCard from "./index";

const mockData = {
  title: "Doctor",
  name: "Jenny Wilson",
  portraitImg: "doctorImg",
  phoneNumber: "+1-340-5558-235 ",
  address: "9094 Bay Meadows Street, Conyers, GA 30012",
  moreInfo:
    "In oculis quidem faciunt, ut dolore magnam aliquam causam ista, quae sine causa? quae fuerit causa, nollem me ab eo delectu rerum, quem modo ista sis",
};

it("renders correctly and matches the snapshot", () => {
  const tree = renderer
    .create(<SeekHelpSingleCard cardDetails={mockData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
