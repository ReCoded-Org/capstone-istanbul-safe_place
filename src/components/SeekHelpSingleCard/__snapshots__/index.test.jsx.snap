// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="carouselItem"
>
  <div
    className="carouselHeaderAndImage"
  >
    <div>
      <h3 />
      <h4 />
    </div>
    <div>
      <img
        alt="Specialist"
        src="defaultImg.svg"
      />
    </div>
  </div>
  <div>
    <div
      className="contactDetails"
    >
      <div>
        <img
          alt="Phone Icon"
          src="phone.svg"
        />
      </div>
      <div>
        <p />
      </div>
    </div>
    <div
      className="contactDetails"
    >
      <div>
        <img
          alt="Location Icon"
          src="map-marker.svg"
        />
      </div>
      <div>
        <p />
      </div>
    </div>
    <p />
  </div>
</div>
`;
